### PR TITLE
Re-did pre-commit-config check

### PR DESCRIPTION
**Pull request by evaherrada**
The check now compares the files from cookecutter with the file in each repo

### CODE REVIEW

**Comment by tekktrik** [comment]
If I understand how this works, this will conflict with https://github.com/adafruit/cookiecutter-adafruit-circuitpython/pull/186 which adds templated content to the YAML file.

**Comment by tekktrik** [comment]
Both - mostly the inner ones, but also changes the upper one to remove the YAML check pre-commit hook because the inner ones aren't valid YAML files anymore

**Comment by tekktrik** [comment]
Never mind, it's still an invalid file per the pre-commit hook at least

**Comment by evaherrada** [comment]
Wait so are you saying it's all good and this won't conflict with the PR?

Also FWIW, the files changed in 186:
Screenshot_2022-06-21_14-21-23

**Comment by tekktrik** [comment]
Sorry, it's not that it will conflict during the merge, but that the solution here means we can't template the file for cookiecutter as 186 does.  The YAML check will flag the YAML files after merging 186 because they aren't valid until after cookiecutter has run them.

**Comment by evaherrada** [comment]
Ohhh I think I understand what you're saying. Basically that if we do it this way we won't be able to do things that were done in 186 (as in make it so cookiecutter can change the file when it is run) to the cookiecutter .pre-commit-config.yaml, right?

**Comment by tekktrik** [comment]
Yup yup!

**Comment by evaherrada** [comment]
Ok. In that case I think the way to do it would be to parse the file for the versions and set them to variables

**Comment by evaherrada** [comment]
@tekktrik How does it look now? On the first run it parses the file from cookiecutter to get the versions and then does the same to the file in each library

**Comment by evaherrada** [comment]
@tekktrik No, it's specifically looking at this one https://github.com/adafruit/cookiecutter-adafruit-circuitpython/blob/main/%7B%7B%20cookiecutter%20and%20'tmp_repo'%20%7D%7D/.pre-commit-config.yaml

**Comment by tekktrik** [comment]
Got it, since that isn't templated it should be good!

**Comment by dhalbert** [comment]
@tekktrik is this good to approve now? Not sure what state it's in.

File: adabot/lib/circuitpython_library_validators.py [hunk 1]
```diff
@@ -240,6 +236,33 @@ def rtd_yml_base(self):
...
 
+    @property
+    def pcc_versions(self):
+        """The parsed YAML from `.pre-commit-config.yaml` in cookiecutter.
+        Used to verify that a library's `.pre-commit-config.yaml` matches this.
+        """
+        if not self._pcc_versions:
+            pcc_yml_dl_url = (
+                "https://raw.githubusercontent.com/adafruit/cookiecutter-adafruit-"
+                "circuitpython/main/%7B%7B%20cookiecutter%20and%20'tmp_repo'%20%7D"
```
Comment (tekktrik) [inline adabot/lib/circuitpython_library_validators.py:247]: This needs to be updated to the new link based on @jepler's modifications

**Review by tekktrik — CHANGES_REQUESTED** [review]
Thanks for pinging @dhalbert, one small change needed.

**Comment by tekktrik** [comment]
Drafting so this doesn't get merged before https://github.com/adafruit/cookiecutter-adafruit-circuitpython/pull/212

**Review by kattni — APPROVED** [review]
Looks good to me, did not test.

**Pull request merged**